user_guide_src/source/changelogs/v4.0.4.rst

```diff
@@ -1,11 +1,59 @@
-Version |version|
+Version 4.0.4
 ====================================================
 
-Release Date: Not released
+Release Date: July 15, 2020
 
-**Next release of CodeIgniter4**
+**4.0.4 release of CodeIgniter4**
 
-Enhancements
-------------
+Enhancements:
+
+- The location for the SQLite3 database has changed and by default will be now located in a ``writable`` folder instead of the ``public`` folder.
+- New CLI command: ``cache:clear`` that will destroy all data within the current cache engine.
+- Thanks to samsonasik, we have several libraries brought up to 100% test coverage.
+- A few small performance increases. Every ounce matters!
+- ``getFieldData()`` in the database results class now returns column type and length.
+- Default .htaccess file updated to work better with extended characters.
+- New Testing Feature: :doc:`Fabricator </testing/fabricator>` makes creating mock classes simple and repeatable in your tests.
+- Model class can now have the callbacks overridden at runtime. Useful for testing.
+- A number of improvements to :doc:`Feature Tests </testing/feature>` in general.
+- New :doc:`command() helper function </cli/cli_commands>` to programatically run your CLI commands. Useful for testing and cron jobs.
+- New command, ``make:seeder`` to generate a :doc:`Database Seed class </dbmgmt/seeds>` skeleton file.
+- Colors now available on the CLI within Windows, as well as other Windows-related CLI improvements.
+- New helper :doc:`mb_url_title </helpers/url_helper>` that functions like ``url_title`` but automatically escapes and extended URL characters.
+- :doc:`Image library </libraries/images>` now supports ``webp`` images.
+- Added Unicode support for regular expressions in the Router.
+- Added support for removing hidden folders in the :doc:`delete_files </helpers/files>` helper
```

Comment:
this should be `delete_files </helpers/filesystem_helper>`

Reply:
Thanks.